src/webgpu/web_platform/util.ts

```diff
@@ -13,113 +13,284 @@ declare global {
   }
 }
 
-export const kVideoInfo =
-  /* prettier-ignore */ makeTable(
-                                             ['mimeType'                     ] as const,
-                                             [undefined                      ] as const, {
-    // All video names
-    'four-colors-vp8-bt601.webm':            ['video/webm; codecs=vp8'       ],
-    'four-colors-theora-bt601.ogv':          ['video/ogg; codecs=theora'     ],
-    'four-colors-h264-bt601.mp4':            ['video/mp4; codecs=avc1.4d400c'],
-    'four-colors-vp9-bt601.webm':            ['video/webm; codecs=vp9'       ],
-    'four-colors-vp9-bt709.webm':            ['video/webm; codecs=vp9'       ],
-    'four-colors-vp9-bt2020.webm':           ['video/webm; codecs=vp9'       ],
-    'four-colors-h264-bt601-rotate-90.mp4':  ['video/mp4; codecs=avc1.4d400c'],
-    'four-colors-h264-bt601-rotate-180.mp4': ['video/mp4; codecs=avc1.4d400c'],
-    'four-colors-h264-bt601-rotate-270.mp4': ['video/mp4; codecs=avc1.4d400c'],
-  } as const);
-export type VideoName = keyof typeof kVideoInfo;
+// MAINTENANCE_TODO: Uses raw floats as expectation in external_texture related cases has some diffs.
+// Remove this conversion utils and uses raw float data as expectation in external_textrue
+// related cases when resolve this.
+export function convertToUnorm8(expectation: {
+  R: number;
+  G: number;
+  B: number;
+  A: number;
+}): Uint8Array {
+  const rgba8Unorm = new Uint8ClampedArray(4);
+  rgba8Unorm[0] = Math.round(expectation.R * 255.0);
+  rgba8Unorm[1] = Math.round(expectation.G * 255.0);
+  rgba8Unorm[2] = Math.round(expectation.B * 255.0);
+  rgba8Unorm[3] = Math.round(expectation.A * 255.0);
 
+  return new Uint8Array(rgba8Unorm.buffer);
+}
+
+// MAINTENANCE_TODO: Add document to describe how to get these magic numbers.
 // Expectation values about converting video contents to sRGB color space.
 // Source video color space affects expected values.
 // The process to calculate these expected pixel values can be found:
 // https://github.com/gpuweb/cts/pull/2242#issuecomment-1430382811
 // and https://github.com/gpuweb/cts/pull/2242#issuecomment-1463273434
 const kBt601PixelValue = {
-  red: new Float32Array([0.972945567233341, 0.141794376683341, -0.0209589916711088, 1.0]),
-  green: new Float32Array([0.248234279433399, 0.984810378661784, -0.0564701319494314, 1.0]),
-  blue: new Float32Array([0.10159735826538, 0.135451122863674, 1.00262982899724, 1.0]),
-  yellow: new Float32Array([0.995470750775951, 0.992742114518355, -0.0774291236205402, 1.0]),
+  displayP3: {
+    red: { R: 0.894192705847254, G: 0.243671200178642, B: 0.132706713001254, A: 1.0 },
+    green: { R: 0.498521961972036, G: 0.971040256815059, B: 0.286694021461098, A: 1.0 },
+    blue: { R: 0.108297853706575, G: 0.134448468828998, B: 0.962798268696141, A: 1.0 },
+    yellow: { R: 0.994991952943137, G: 0.992827230430229, B: 0.318417391806334, A: 1.0 },
+  },
```

Comment:
We have the `srgbToDisplayP3` helper already (helpfully copied from the CSS Color Level 4 spec). Is it possible to use that here, so we only have to maintain one copy of the magic numbers?

Reply:
(It would be cool to have helpers for bt601/bt709, so we wouldn't need any magic numbers, but that is not important.)

Reply:
That would be ideal!

Reply:
Cannot find exist color space conversion from BT.601 and BT.709, Left a TODO for removing all magic numbers in future.